scrapy/utils/misc.py

```diff
@@ -35,13 +35,15 @@ def load_object(path):
     path ie: 'scrapy.contrib.downloadermiddelware.redirect.RedirectMiddleware'
     """
 
-    try:
-        dot = path.rindex('.')
-    except ValueError:
-        raise ValueError("Error loading object '%s': not a full path" % path)
-
-    module, name = path[:dot], path[dot+1:]
-    mod = import_module(module)
+    if isinstance(path,str) :
+        if '.' in path :
+            dot = path.rindex('.')
+            module, name = path[:dot], path[dot+1:]
+            mod = import_module(module)
+        else:
+            raise Exception()
```

Comment:
yes the exception should be 
   raise ValueError("Error loading object '%s': not a full path" % path)
 not raise exception.

Here is how I am using it :
https://github.com/flosokaks/scrapy_ks/blob/master/scrapy_openkansas.py#L293

settings.set('ITEM_PIPELINES',{
PythonCache : 300,
MarkDownCache : 300,
HtmlCache : 300,
})